modules/carbonRtdProvider.js

```diff
@@ -0,0 +1,218 @@
+/**
+* This module adds the carbon provider to the Real Time Data module (rtdModule)
+* The {@link module:modules/realTimeData} module is required
+* The module will add contextual and audience targeting data to bid requests
+* @module modules/carbonRtdProvider
+* @requires module:modules/realTimeData
+*/
+import { submodule } from '../src/hook.js';
+import { ajax } from '../src/ajax.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import { logError, isGptPubadsDefined } from '../src/utils.js';
+import { getStorageManager } from '../src/storageManager.js';
+import { config as sourceConfig } from '../src/config.js';
+
+const SUBMODULE_NAME = 'carbon'
+const CARBON_GVL_ID = 493;
+const MODULE_VERSION = 'v1.0'
+const DATA_HOST = 'https://pb-rtd.ccgateway.net'
+const STORAGE_KEY = 'carbon_data'
+const PROFILE_ID_KEY = 'carbon_ccuid'
+
+export const storage = getStorageManager({ gvlid: CARBON_GVL_ID, moduleName: SUBMODULE_NAME })
+
+export function setLocalStorage(carbonData) {
+  let data = JSON.stringify(carbonData);
+  storage.setDataInLocalStorage(STORAGE_KEY, data);
+}
+
+export function matchCustomTaxonomyRule (rule) {
+  const contentText = window.top.document.body.innerText;
+  if (rule.MatchType == 'any') {
+    for (var anyWord in rule.WordWeights) {
+      var anyRegex = RegExp('\\b' + anyWord + '\\b', 'i');
+      var anyResult = contentText.match(anyRegex);
```

Comment:
If there is not a specific reason to use var could we change them all to `let` or `const`